src/main/webapp/dashboard-manager.js

```diff
@@ -15,6 +15,16 @@ class DashboardManager {
     google.visualization.events.addListener(
         this.aggregationSelector, 'statechange', this.updateAndDrawData.bind(this));
 
+    document.addEventListener('bulkUpdate', (e) => {
+      let row = e.detail;
+      let deviceIds = this.data.getValue(row, this.data.getNumberOfColumns() - 2);
```

Comment:
Can `this.data.getNumberOfColumns() - 2` be defined as a constant? It's hard for others who are less familiar with the code to understand why there is a `2` here.

Here and below.